### ISSUE TITLE
Repo structure and development processes

### ISSUE THREAD

**Issue by hammer** [refstudio/refstudio#2]
A good start is in https://github.com/sergioramos/refstudio.

Some decisions:
- React
- Yarn
- Prettier
- TypeScript
- Vite: default for Tauri

Still to be decided:
- Testing and linting? Jest, ESLint, etc.

**Comment by denjell-crabnebula** [comment]
I think there are also a number of good integrations with vscode that you can look at, and you might also consider using devcontainer, depending on your aspirations.

1. react is a good choice because it is well known in the ecosystem and many devs love it

2. today it seems most people working with tauri prefer pnpm, but i am not sure if you will grow into a monorepo style - in which case yarn might be better. whatever you do, make sure it becomes canon 

3. prettier can be great, just be sure to use either githooks or integrate into your github action workflows. that said i have always preferred eslint + standardjs. linting is good because you can add a lot of framework specific foo and add security checks etc. this is a good primer: https://loige.hashnode.dev/configure-eslint-and-standardjs 

4. this is smart, you will at some point want to have typing on your tauri commands, there are a few options here that are best discussed in a chat

5. vite is a great choice

--

Testing is complicated. 
We have heard good things about https://nodejs.org/api/test.html (but jest is what i would have recommended a while ago for unit testing)
However, you will at some point want e2e tests, and in my opinion webdriver.io is probably the best way forward, since we have pretty decent support in tauri with our own tauri-runner that can be integrated on windows and linux (macos is still a wip).

**Comment by hammer** [comment]
Development process note: I prefer to have an issue open before a PR is submitted so that high-level topics can be discussed on the issue and PRs can be for code-level discussion.

**Comment by hammer** [comment]
How do we want our git history to look?

<img width="315" alt="Screenshot 2023-05-18 at 7 58 40 AM" src="https://github.com/refstudio/refstudio/assets/15233/1363f35c-0b34-4bc1-826c-7d93b7bb171d">

**Comment by danvk** [comment]
@hammer I'm a fan of "Squash and merge". It gives you a nice, clean, linear history on `master` which makes it easier to bisect and track changes. And it lets you make the history on your PR branches as messy as you like.

**Comment by hammer** [comment]
Yeah I have used squash and merge for most projects. I'm curious if anyone has opinions on https://mergify.com. We're using it at https://github.com/pystatgen/sgkit but I personally don't have a strong opinion on it.

**Comment by cguedes** [comment]
I also a fan of "Squash and merge".  I don't have an opinion about using that external [mergify](https://mergify.com)https://mergify.com/ tool.

**Comment by hammer** [comment]
I'm going to close this for now, we can file additional improvements as new issues.
